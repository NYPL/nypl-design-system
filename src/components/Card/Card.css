@import "../Heading/Heading.css";

@custom-media --card-break (max-width var(--breakpoint-medium));

.card {
  align-items: flex-start;
  border: 1px solid var(--ui-gray-light);
  border-radius: 5px;
  display: flex;
  flex-flow: column wrap;
  margin-bottom: var(--space-s);
  padding: var(--space) var(--space-l);

  &__heading,
  &__footer {
    flex: 100%;
  }

  &__columns {
    display: flex;
    flex-flow: column wrap;
    justify-content: space-between;
    width: 100%;

    @media (--card-break) {
      flex-flow: row wrap;
    }
  }

  &__columns + &__footer {
    margin-top: var(--space-xs);
  }

  &__image {
    align-items: center;
    background-color: var(--ui-gray-xlight);
    display: flex;
    flex: 0 0 150px;
    height: 150px;
    justify-content: center;
    margin-bottom: var(--space-s);

    .image {
      max-height: 100%;
      max-width: 100%;
    }

    @media (--card-break) {
      align-self: center;
      margin-bottom: 0;
      margin-right: var(--space-s);
      width: 150px;
    }
  }

  &__content {
    flex: 1 1;
    margin-bottom: var(--space-s);

    @media (--card-break) {
      margin-bottom: 0;
    }
  }

  &__ctas {
    flex: 0 1 20%;
  }

  &__content + &__ctas {
    margin-right: var(--space-s);
  }
}

/* Stories-only styles */
.sb-show-main {
  .var-docs {
    padding: 0 0 var(--space) 0;
  }

  .example-card {
    .card__content {
      > * {
        margin-bottom: var(--space-xxs);
      }
    }

    .book__callout {
      @add-mixin heading-xs;

      font-style: italic;
      font-weight: 300;
    }

    .card__ctas {
      align-items: flex-end;
      display: flex;
      flex-flow: row wrap;
      justify-content: center;
      text-align: center;

      .button {
        width: 100%;
      }

      > * {
        margin-bottom: var(--space-xxs);
      }
    }

    .book__filetype {
      margin-bottom: var(--space-s);
    }

    .book__description {
      font-style: italic;
    }
  }
}
