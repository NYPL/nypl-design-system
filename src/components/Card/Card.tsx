import * as React from "react";
import bem from "../../utils/bem";

interface CardProps {
  /** ClassName that appears in addition to "card" */
  className?: string;
  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** Calls to Action for the card, such as "download" on a card for a book */
  ctas?: React.ReactNode;
  /** Footer that appears within the card but below the three columns within it */
  footer?: React.ReactNode;
  /** Heading that appears within the card but above the three columns within it */
  heading?: React.ReactNode;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
  /** First column within the card */
  image?: React.ReactNode;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
}

export default function Card(props: React.PropsWithChildren<CardProps>) {
  const {
    blockName,
    children,
    className,
    ctas,
    footer,
    heading,
    id,
    image,
    modifiers = [],
  } = props;
  const baseClass = "card";

  return (
    <div className={bem(baseClass, modifiers, blockName, [className])} id={id}>
      {heading && (
        <div className={bem("heading", modifiers, baseClass)}>{heading}</div>
      )}
      <div className={bem("columns", modifiers, baseClass)}>
        {image && (
          <div className={bem("image", modifiers, baseClass)}>{image}</div>
        )}
        {children && (
          <div className={bem("content", modifiers, baseClass)}>{children}</div>
        )}
        {ctas && (
          <div className={bem("ctas", modifiers, baseClass)}>{ctas}</div>
        )}
      </div>
      {footer && (
        <div className={bem("footer", modifiers, baseClass)}>{footer}</div>
      )}
    </div>
  );
}
