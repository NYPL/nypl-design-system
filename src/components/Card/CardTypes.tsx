export enum CardLayouts {
  Horizontal = "horizontal",
  Vertical = "vertical",
}
