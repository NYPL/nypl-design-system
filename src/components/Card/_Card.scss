.card {
  @include space-stack-s;

  align-items: flex-start;
  border: 1px solid var(--ui-gray-light);
  border-radius: 5px;
  display: flex;
  flex-flow: column wrap;
  padding: var(--space) var(--space-l);

  &__heading,
  &__footer {
    flex: 100%;
  }

  &__columns {
    display: flex;
    flex-flow: column wrap;
    justify-content: space-between;
    width: 100%;

    @include breakpoint($breakpoint-medium) {
      flex-flow: row wrap;
    }
  }

  &__columns + &__footer {
    margin-top: var(--space-xs);
  }

  &__image {
    @include space-stack-s;

    align-items: center;
    background-color: var(--ui-gray-xlight);
    display: flex;
    flex: 0 0 150px;
    height: 150px;
    justify-content: center;

    .image {
      max-height: 100%;
      max-width: 100%;
    }

    @include breakpoint($breakpoint-medium) {
      @include space-stack-none;
      @include space-inline-s;

      align-self: center;
      width: 150px;
    }
  }

  &__content {
    @include space-stack-s;

    flex: 1 1;

    @include breakpoint($breakpoint-medium) {
      @include space-inline-s;
      @include space-stack-none;
    }
  }

  &__ctas {
    flex: 0 1 20%;

    @include breakpoint($breakpoint-medium) {
      @include space-inline-s;
    }
  }

  &__content + &__ctas {
    @include space-inline-s;
  }
}

// Stories-only styles
.sb-show-main {
  .var-docs {
    padding: 0 0 var(--space) 0;
  }

  .example-card {
    .card__content {
      > * {
        @include space-stack-xxs;
      }
    }

    .book__callout {
      @include heading-xs;

      font-style: italic;
      font-weight: 300;
    }

    .card__ctas {
      align-items: flex-end;
      display: flex;
      flex-flow: row wrap;
      justify-content: center;
      text-align: center;

      .button {
        width: 100%;
      }

      > * {
        @include space-stack-xxs;
      }
    }

    .book__filetype {
      @include space-stack-s;
    }

    .book__description {
      font-style: italic;
    }
  }
}
