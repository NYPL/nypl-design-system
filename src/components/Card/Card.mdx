import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";

import * as CardStories from "./Card.stories";
import Link from "../Link/Link";
import { changelogData } from "./cardChangelogData";
import ComponentChangelogTable from "../../utils/ComponentChangelogTable";

<Meta of={CardStories} />

# Card

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.24.0`   |
| Latest            | `3.1.2`    |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#image-position" target="_self">Image Position</Link>}
- {<Link href="#image-size" target="_self">Image Size</Link>}
- {<Link href="#custom-image-component" target="_self">Custom Image Component</Link>}
- {<Link href="#card-with-link-heading" target="_self">Card With Link Heading</Link>}
- {<Link href="#card-with-full-click-functionality" target="_self">Card With Full-Click Functionality</Link>}
- {<Link href="#card-with-right-side-cardactions" target="_self">Card with Right Side CardActions</Link>}
- {<Link href="#cards-in-a-grid" target="_self">Cards in a Grid</Link>}
- {<Link href="#cards-in-a-stack" target="_self">Cards in a Stack</Link>}
- {<Link href="#cards-without-images" target="_self">Cards Without Images</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={CardStories} />

The `Card` component is viewed as a contained, stand-alone element. It is highly
configurable and composable. It can be shown with or without an image and the
content area is composed using Card-specific child components: `CardHeading`,
`CardActions` and `CardContent`. These child components can be arranged in any
order and each child component may be used multiple times within the content
area. Although a `Card` is a self-contained unit, `Card` components are often
grouped together and displayed in some type of grid layout.

Note: all of the examples have images with the `isLazy` set to true.

### CardHeading

The `CardHeading` component mirrors the standard Reservoir Design System (DS)
`Heading` component and accepts the [same props](../?path=/story/components-typography-styles-heading--with-controls).

### CardActions

The `CardActions` component is used to display "call to action" (CTA) buttons
and links for the card. The DS `Button` and `Link` components should be passed
as children into the `CardActions` component.

Set the `isAlignedRightActions` prop to true in the `Card` component to align the
`CardActions` to the right of the the main content area. This is only applicable
in the row layout of the `Card` component.

### CardContent

The `CardContent` component should be used to display all content other than
headings and CTAs.

### CardImage

The `CardImage` component is _not_ exported and is used internally only in the
`Card` component. It is a convenience component that mirrors the standard DS
`Image` component. This means that the `CardImage` component accepts the same
props as the `Image` component, in addition to the `isAtEnd` prop.

Developers do _not_ need to implement this component. Instead, it will be
rendered in the `Card` component through its `imageProps` prop.

<Source
  code={`
<Card
  imageProps={{
    alt: "Alt text",
    aspectRatio: "twoByOne",
    caption: "caption text",
    credit: "credit text",
    isAtEnd: true,
    isLazy: true,
    size: "medium",
    src: "//loremflickr.com/400/200/new+york+public+library",
  }}
  {...otherCardProps}
>
  <CardHeading level="h3" id="heading-id">
    Heading Text
  </CardHeading>
  <CardContent>
    Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
    Sed posuere consectetur est at lobortis. Cras justo odio, dapibus ac
    facilisis in, egestas eget quam.
  </CardContent>
</Card>
`}
  language="jsx"
/>

## Component Props

<Canvas of={CardStories.WithControls} />

<ArgTypes of={CardStories.WithControls} />

## Accessibility

### Headings

In general, headings from `h1` through `h6` must be in sequential order in the
DOM tree. Rendering an `h4` following an `h2` will throw an accessibility error.
Therefore, when adding `CardHeading` components, make sure to add them in
sequential order with the proper `level` prop.

<Source
  code={`
// Correct

<CardHeading level="h3" id="headingthree">
  Heading level three
</CardHeading>
<CardHeading level="h4" id="headingfour">
  Heading level four
</CardHeading>

// Incorrect

<CardHeading level="h2" id="headingtwo">
  Heading level Two
</CardHeading>
<CardHeading level="h4" id="headingfour">
  Heading level four
</CardHeading>
`} language="jsx" />

### Full-Click Functionality

Passing in a URL in the `mainActionLink` prop will make the entire `Card`
clickable. Other links in the `CardActions` component can still be accessed by
tabbing through and pressing "enter" or by clicking as you normally would on a
link.

## Image Position

By default, the image is always the first element within a `Card`. This means
the image is at the top of a `Card` with a column layout and on the left side of
a `Card` with a row layout. However, you can use the `imageProps.isAtEnd` boolean
prop to override the default placement and move the image to the last element
within a `Card`.

### Column Cards

<Canvas of={CardStories.ImagePositionColumnCards} />

### Row Cards

<Canvas of={CardStories.ImagePositionRowCards} />

## Image Size

By default, the image width is `100%` for a `Card` with a column layout and
`225px` for a `Card` with a row layout. However, you can use the `imageProps.size`
prop to override the default width.

### Column Cards

<Canvas of={CardStories.ImageSizeColumnCards} />

### Row Cards

<Canvas of={CardStories.ImageSizeRowCards} />

## Custom Image Component

Instead of passing a path for the `imageProps.src` prop, you can pass a custom
image component in using the `imageProps.component` prop.

<Canvas of={CardStories.CustomImageComponent} />

## Card with Link Heading

The `CardHeading` component works just like the DS `Heading` component. This
means that a URL can be passed in the `url` prop for `CardHeading` or a `Link`
component can be used as a child. Click on "Show code" in the example below to
see both patterns.

<Canvas of={CardStories.HeadingAsLink} />

## Card With Full-Click Functionality

To enable the full-click functionality in the `Card` component, pass a URL in
the `mainActionLink` prop. This will make the entire `Card` component clickable.
If the `Card` has action links or buttons, they can still be accessed by tabbing
through the `Card` and pressing `enter` or clicking like a regular link.

Internally, if multiple `CardHeading` components are passed, only the first one
will have the full-click link. This, however, does not affect the `Card` itself
from being having the full-click functionality.

<Canvas of={CardStories.FullClick} />

This example can be found in the [Turbine homepage](https://nypl-ds-test-app.vercel.app/).

<Canvas of={CardStories.CardFullClickTurbineExample} />

## Card With Full-Click Functionality and Tooltip

<Canvas of={CardStories.FullClickWithTooltip} />

## Card with Right Side CardActions

It's possible to set only the `CardActions` component on the right side of the
main content area of the `Card`. This is possible only in the `layout="row"`
layout and through the `Card` component's `isAlignedRightActions` prop which
must be set to `true`.

<Canvas of={CardStories.CardWithRightSideCardActions} />

## Cards in a Grid

<Canvas of={CardStories.GridExample} />

## Cards in a Stack

<Canvas of={CardStories.StackExample} />

## Cards Without Images

<Canvas of={CardStories.WithoutImages} />

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
