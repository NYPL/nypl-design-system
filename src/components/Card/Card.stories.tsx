import * as React from "react";
import { action } from "@storybook/addon-actions";
import { text, boolean, select } from "@storybook/addon-knobs";

import Card from "./Card";
import Heading from "../Heading/Heading";
import Image from "../Image/Image";
import Button from "../Button/Button";
import Link from "../Link/Link";
import Icon from "../Icons/Icon";
import { IconRotationTypes, IconNames } from "../Icons/IconTypes";
import { LinkTypes } from "../Link/LinkTypes";
import { ButtonTypes } from "../Button/ButtonTypes";

export default {
  title: "Card",
  component: Card,
};

let showCTAs, showImage, showHeader, showFooter;

const headerLevels = [1, 2, 3, 4, 5, 6];

const imageRatios = {
  "1x1": "https://placeimg.com/100/100/animals",
  "2x1": "https://placeimg.com/200/100/animals",
  "3x4": "https://placeimg.com/150/200/animals",
  "4x1": "https://placeimg.com/400/100/animals",
  "4x3": "https://placeimg.com/200/150/animals",
  "16x9": "https://placeimg.com/400/225/animals",
};

export const card = () => (
  <>
    {boolean("Show Header", true) ? (showHeader = true) : (showHeader = false)}
    {boolean("Show Image", true) ? (showImage = true) : (showImage = false)}
    {boolean("Show CTAs", true) ? (showCTAs = true) : (showCTAs = false)}
    {boolean("Show Footer", true) ? (showFooter = true) : (showFooter = false)}
    <Card
      id="cardID"
      heading={
        showHeader ? (
          <Heading
            level={select("Header Level", headerLevels, headerLevels[2])}
            id="heading1"
            text={text("Header Content", "Optional Header")}
          />
        ) : (
          false
        )
      }
      image={
        showImage ? (
          <Image
            src={select("Image Ratio", imageRatios, imageRatios[2])}
            alt={""}
          />
        ) : null
      }
      ctas={
        showCTAs ? (
          <Button
            onClick={action("clicked")}
            id="button1"
            buttonType={ButtonTypes.Primary}
            type="submit"
          >
            {text("CTA Button Text: ", "Example CTA")}
          </Button>
        ) : null
      }
      footer={
        showFooter ? <>{text("Footer content: ", "Optional footer")}</> : null
      }
    >
      {text("Card Content: ", "Middle column content")}
    </Card>
  </>
);

card.storyName = "Card";

card.parameters = {
  design: {
    type: "figma",
    url:
      "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=17167%3A58131",
  },
};

export const ExampleCard = () => (
  <Card
    className="example-card"
    image={<Image src="https://placeimg.com/300/400/arch" alt={""} />}
    ctas={
      <>
        <Link type={LinkTypes.Button} href="blah">
          Reserve
        </Link>
        <div className="italicized ui-gray-dark">
          0 of 11 copies available. 3 patrons in the queue.
        </div>
        <Link href="#url" type={LinkTypes.Forwards}>
          View Book Details
        </Link>
      </>
    }
  >
    <>
      <Heading level={2} id="heading1" text={"The Year of Magical Thinking"} />
      <div className="book__callout">A portrait of loss and grief</div>
      <div>
        By <Link href="#joan-didion">Joan Didion</Link>,{" "}
        <Link href="#another-author">Another Author</Link>,{" "}
        <Link href="#third-author">Third Author</Link>
      </div>
      <div className="book__filetype">
        <Icon
          name={IconNames.headset}
          decorative={true}
          modifiers={["left", "small"]}
          iconRotation={IconRotationTypes.rotate0}
        ></Icon>
        Audiobook
      </div>
      <div className="book__description">
        An act of consummate literary bravery, a writer known for her clarity
        allowing us to watch her mind as it becomes clouded...{" "}
        <Link href="#">Read More</Link>
      </div>
    </>
  </Card>
);

ExampleCard.storyName = "Example Card";

ExampleCard.parameters = {
  design: {
    type: "figma",
    url:
      "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=17167%3A58131",
  },
};
