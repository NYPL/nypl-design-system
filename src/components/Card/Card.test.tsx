import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";

import Card from "./Card";
import Link from "../Link/Link";
import { LinkTypes } from "../Link/LinkTypes";
import Heading from "../Heading/Heading";
import Image from "../Image/Image";
import Button from "../Button/Button";
import { ButtonTypes } from "../Button/ButtonTypes";
import Icon from "../Icons/Icon";
import { IconRotationTypes, IconNames } from "../Icons/IconTypes";

describe("Card", () => {
  const regularCard = (
    <Card
      id="cardID"
      heading={<Heading level={3} id="heading1" text={"Optional Header"} />}
      image={<Image src="https://placeimg.com/400/200/arch" alt={""} />}
      ctas={
        <Button
          onClick={function () {
            console.log(this);
          }}
          id="button1"
          buttonType={ButtonTypes.Primary}
          type="submit"
        >
          Example CTA
        </Button>
      }
      footer={<>Optional footer</>}
    >
      middle column content
    </Card>
  );

  const cardWithExtendedStyles = (
    <Card
      id="card#1"
      className="edition-card"
      heading={
        <Heading level={2} id="editioncardheading1" text={"2004 Edition"} />
      }
      image={<Image src="https://placeimg.com/300/400/arch" alt={""} />}
      ctas={
        <div className="edition-card__ctas">
          <Link type={LinkTypes.Button} href="blah">
            Read Online
          </Link>
          <div className="edition-card__download">
            <Link href="#url" type={LinkTypes.Action}>
              <Icon
                name={IconNames.download}
                blockName="more-link"
                decorative={true}
                modifiers={["left"]}
                iconRotation={IconRotationTypes.rotate0}
              ></Icon>
              Download
            </Link>
          </div>
        </div>
      }
    >
      <>
        <div>Published in New York by Random House</div>
        <div>Written in English</div>
        <div>
          License: Creative Commons Attribution-NonCommercial-NoDerivatives 4.0
          International
        </div>
      </>
    </Card>
  );

  const cardWithNoCTAs = (
    <Card
      id="card#1"
      image={<Image src="https://placeimg.com/300/400/arch" alt={""} />}
      className="edition-card"
      heading={
        <Heading level={2} id="editioncardheading1" text={"2004 Edition"} />
      }
    >
      <>
        <div>Published in New York by Random House</div>
        <div>Written in English</div>
        <div>Under Creative Commons License</div>
      </>
    </Card>
  );

  const cardWithNoContent = (
    <Card
      id="card#1"
      className="edition-card"
      image={<Image src="https://placeimg.com/300/400/arch" alt={""} />}
      heading={
        <Heading
          level={2}
          id="editioncardheading1"
          url="#edition-link"
          text={"2004 Edition"}
        />
      }
      ctas={
        <div className="edition-card__ctas">
          <Link type={LinkTypes.Button} href="blah">
            Read Online
          </Link>
          <div className="edition-card__download">
            <Link href="#url" type={LinkTypes.Action}>
              <Icon
                name={IconNames.download}
                blockName="more-link"
                decorative={true}
                modifiers={["left"]}
                iconRotation={IconRotationTypes.rotate0}
              ></Icon>
              Download
            </Link>
          </div>
        </div>
      }
    ></Card>
  );

  const cardWithNoImage = (
    <Card
      id="card#1"
      className="edition-card"
      heading={
        <Heading
          level={2}
          id="editioncardheading1"
          url="#edition-link"
          text={"2004 Edition"}
        />
      }
      ctas={
        <div className="edition-card__ctas">
          <Link type={LinkTypes.Button} href="blah">
            Read Online
          </Link>
          <div className="edition-card__download">
            <Link href="#url" type={LinkTypes.Action}>
              <Icon
                name={IconNames.download}
                blockName="more-link"
                decorative={true}
                modifiers={["left"]}
                iconRotation={IconRotationTypes.rotate0}
              ></Icon>
              Download
            </Link>
          </div>
        </div>
      }
    >
      middle column content
    </Card>
  );

  it("Generates a Card with a header, footer, image, middle content, and CTAs", () => {
    const card = Enzyme.mount(regularCard);
    expect(card.find(".card__heading")).to.have.lengthOf(1);
    expect(card.find(".card__image")).to.have.lengthOf(1);
    expect(card.find(".card__content")).to.have.lengthOf(1);
    expect(card.find(".card__ctas")).to.have.lengthOf(1);
    expect(card.find(".card__footer")).to.have.lengthOf(1);
  });

  it("Generates a Card with variable data", () => {
    const card = Enzyme.mount(cardWithExtendedStyles);
    expect(card.find("h2")).to.have.lengthOf(1);
    expect(card.find(".card__content").find("div")).to.have.lengthOf(4);
    expect(card.find(".card__ctas").find("a")).to.have.lengthOf(2);
  });

  it("Generates a card without a CTA block if one isn't provided", () => {
    const card = Enzyme.mount(cardWithNoCTAs);
    expect(card.find(".card__heading")).to.have.lengthOf(1);
    expect(card.find(".card__image")).to.have.lengthOf(1);
    expect(card.find(".card__content")).to.have.lengthOf(1);
    expect(card.find(".card__ctas")).to.have.lengthOf(0);
  });

  it("Generates a card without a content block if one isn't provided", () => {
    const card = Enzyme.mount(cardWithNoContent);
    expect(card.find(".card__heading")).to.have.lengthOf(1);
    expect(card.find(".card__image")).to.have.lengthOf(1);
    expect(card.find(".card__content")).to.have.lengthOf(0);
    expect(card.find(".card__ctas")).to.have.lengthOf(1);
  });

  it("Generates a card without an image block if no image is provided", () => {
    const card = Enzyme.mount(cardWithNoImage);
    expect(card.find(".card__heading")).to.have.lengthOf(1);
    expect(card.find(".card__image")).to.have.lengthOf(0);
    expect(card.find(".card__content")).to.have.lengthOf(1);
    expect(card.find(".card__ctas")).to.have.lengthOf(1);
  });
});
