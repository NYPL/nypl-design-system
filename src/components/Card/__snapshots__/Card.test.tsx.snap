// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders the UI snapshot correctly 1`] = `
<div
  className="css-1u8qly9"
  id="regularCard"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/400/200/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h3
      className="chakra-heading css-1xdhyk6"
      id="heading1"
      layout="column"
    >
      The Card Heading
    </h3>
    <div
      className="css-1xdhyk6"
    >
      middle column content
    </div>
    <div
      className="css-1xdhyk6"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button1"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 2`] = `
<div
  className="edition-card css-1u8qly9"
  id="cardWithExtendedStyles"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/300/400/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h2
      className="chakra-heading css-1xdhyk6"
      id="editioncardheading1"
      layout="column"
    >
      The Card Heading
    </h2>
    <div
      className="css-1xdhyk6"
    >
      <div>
        Published in New York by Random House
      </div>
      <div>
        Written in English
      </div>
      <div>
        License: Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International
      </div>
    </div>
    <div
      className="css-1xdhyk6"
    >
      <a
        className="css-1xdhyk6"
        href="online"
        id="link-online"
        rel={null}
        target={null}
      >
        Read Online
      </a>
      <a
        className="css-1xdhyk6"
        href="#url"
        id="link-icon"
        rel={null}
        target={null}
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-1grhd2q"
          focusable={false}
          id="icon-cardWithExtendedStyles"
          role="img"
          title="download icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        Download
      </a>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 3`] = `
<div
  className="edition-card css-1u8qly9"
  id="cardWithNoCTAs"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/300/400/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h2
      className="chakra-heading css-1xdhyk6"
      id="editioncardheading1"
      layout="column"
    >
      The Card Heading
    </h2>
    <div
      className="css-1xdhyk6"
    >
      <p>
        Published in New York by Random House
      </p>
      <p>
        Written in English
      </p>
      <p>
        Under Creative Commons License
      </p>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 4`] = `
<div
  className="edition-card css-1u8qly9"
  id="cardWithNoContent"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/300/400/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h2
      className="chakra-heading css-1xdhyk6"
      id="editioncardheading1"
      layout="column"
    >
      <a
        className="css-1xdhyk6"
        href="#edition-link"
        id="editioncardheading1-link"
        rel={null}
        target={null}
      >
        The Card Heading
      </a>
    </h2>
    <div
      className="css-1xdhyk6"
    >
      <a
        className="css-1xdhyk6"
        href="online"
        id="link-online"
        rel={null}
        target={null}
      >
        Read Online
      </a>
      <a
        className="css-1xdhyk6"
        href="#url"
        id="link-icon"
        rel={null}
        target={null}
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-1grhd2q"
          focusable={false}
          id="icon-cardWithNoContent"
          role="img"
          title="download icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        Download
      </a>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 5`] = `
<div
  className="edition-card css-1u8qly9"
  id="cardWithNoImage"
>
  <div
    className="card-body css-0"
  >
    <h2
      className="chakra-heading css-1xdhyk6"
      id="editioncardheading1"
      layout="column"
    >
      <a
        className="css-1xdhyk6"
        href="#edition-link"
        id="editioncardheading1-link"
        rel={null}
        target={null}
      >
        The Card Heading
      </a>
    </h2>
    <div
      className="css-1xdhyk6"
    >
      middle column content
    </div>
    <div
      className="css-1xdhyk6"
    >
      <a
        className="css-1xdhyk6"
        href="online"
        id="link-online"
        rel={null}
        target={null}
      >
        Read Online
      </a>
      <a
        className="css-1xdhyk6"
        href="#url"
        id="link-icon"
        rel={null}
        target={null}
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-1grhd2q"
          focusable={false}
          id="icon-cardWithNoImage"
          role="img"
          title="download icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        Download
      </a>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 6`] = `
<div
  className="chakra-linkbox css-113b4ge"
  id="fullclick"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/400/200/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h3
      className="chakra-heading css-1xdhyk6"
      id="heading1"
      layout="column"
    >
      <a
        className="chakra-linkbox__overlay css-kvj8dg"
        href="http://nypl.org"
      >
        The Card Heading
      </a>
    </h3>
    <div
      className="css-1xdhyk6"
    >
      middle column content
    </div>
    <div
      className="css-1xdhyk6"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button1"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 7`] = `
<div
  className="css-1u8qly9"
  id="cardID"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/400/200/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h3
      className="chakra-heading css-1xdhyk6"
      id="heading1"
      layout="column"
    >
      The Card Heading
    </h3>
    <div
      className="css-1xdhyk6"
    >
      middle column content
    </div>
    <div
      className="css-1xdhyk6"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button1"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
    <div
      className="css-1xdhyk6"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button2"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 8`] = `
<div
  className="css-2bbtg1"
  id="chakraProps"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/400/200/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h3
      className="chakra-heading css-1ss7kwh"
      id="heading1"
      layout="column"
    >
      The Card Heading
    </h3>
    <div
      className="css-1rn2qcq"
    >
      middle column content
    </div>
    <div
      className="css-1avkygu"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button1"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
  </div>
</div>
`;

exports[`Card Renders the UI snapshot correctly 9`] = `
<div
  className="css-1u8qly9"
  data-testid="card-testid"
  id="otherProps"
>
  <div
    className="css-0"
  >
    <div
      className="css-0"
      data-testid="test"
    >
      <img
        alt="Alt text"
        className="css-0"
        src="https://placeimg.com/400/200/arch"
      />
    </div>
  </div>
  <div
    className="card-body css-0"
  >
    <h3
      className="chakra-heading css-1ss7kwh"
      id="heading1"
      layout="column"
    >
      The Card Heading
    </h3>
    <div
      className="css-1xdhyk6"
      data-testid="cardcontent-testid"
    >
      middle column content
    </div>
    <div
      className="css-1xdhyk6"
      data-testid="cardaction-testid"
    >
      <button
        className="chakra-button css-1xdhyk6"
        id="button1"
        onClick={[Function]}
        type="submit"
      >
        Example CTA
      </button>
    </div>
  </div>
</div>
`;
